ta/os_test/os_test.c

```diff
@@ -635,6 +631,10 @@ static TEE_Result test_time(void)
 	printf("TA time %u.%03u\n", (unsigned int)t.seconds,
 	       (unsigned int)t.millis);
 
+	TEE_GetSystemTime(&sys_t);
+	printf("system time %u.%03u\n", (unsigned int)sys_t.seconds,
+	       (unsigned int)sys_t.millis);
+
 	if (t.seconds > sys_t.seconds) {
```

Comment:
This test seems a bit strange. What are we testing against here, a time that has wrapped too much?

Reply:
I guess this test checks if the TA time wrapped correctly?
If the TA time wraps correctly after the one second delay, the TA time should be inferior to one second. That TA time will always be inferior to the system time (with that patch).
If the TA time does not wrap and stays to its max value, the TA time will be superior to system time. There is still a chance where TA time and system time are at the maximum value at the very same time ... but that's unlikely.

Reply:
I guess the test checks that TA time has indeed wrapped. Maybe testing against 0 should be sufficient.